internal/planner/planner.go

```diff
@@ -252,10 +252,18 @@ func getScaleDeployments(
 		}
 
 		switch version.Status {
-		case temporaliov1alpha1.VersionStatusInactive,
-			temporaliov1alpha1.VersionStatusRamping,
-			temporaliov1alpha1.VersionStatusCurrent:
-			// TODO(carlydf): Consolidate scale up cases and verify that scale up is the correct action for inactive versions
+		case temporaliov1alpha1.VersionStatusInactive:
+			// Scale down inactive versions that are not the target
+			if status.TargetVersion.BuildID == version.BuildID {
+				if d.Spec.Replicas != nil && *d.Spec.Replicas != replicas {
+					scaleDeployments[version.Deployment] = uint32(replicas)
+				}
+			} else {
+				if d.Spec.Replicas != nil && *d.Spec.Replicas != 0 {
+					scaleDeployments[version.Deployment] = 0
```

Comment:
This would scale down workers where a test workflow failed and therefore the version never became active. We've discussed this a few times, just wanted to make sure we've considered that here. It means we will lose the logs.

Reply:
But only _after_ a new target version has been configured, correct? That behavior makes sense to me because you still have the ability to grab logs via kubectl before rolling forward/back.

Reply:
yes, only when there is a new target version.